source/OutfitInfoDisplay.cpp

```diff
@@ -192,6 +192,13 @@ namespace {
 		{"minable", "This item is mined from asteroids."},
 		{"atrocity", "This outfit is considered an atrocity."}
 	};
+
+	bool IsNotRequirement(const string& label)
+	{
+		return label == "automaton" ||
+			SCALE.find(label) != SCALE.end() ||
+			BOOLEAN_ATTRIBUTES.find(label) != BOOLEAN_ATTRIBUTES.end();
+	}
```

Comment:
Looking at the scales and booleans seems like a weird way of doing it? I'm not saying it is necessarily wrong, I'm just not sure I understand why it is the case. We could instead use a list of actual requirements?

I see line 348-349 saying `They conveniently don't use SCALE or BOOLEAN_ATTRIBUTES.` which doesn't seem very resilient as a way of handling it.
This function could at least use a good comment in front to explain why it works this way, which idk if it should.


Also, automaton is a boolean ship attribute; so you don't have to consider it as an exception; if at some point it gets added to outfits it'll be a part of `BOOLEAN_ATTRIBUTES`.

Reply:
So I'm looking at this, and re-writing this as "IsRequirement" which also takes the value. This started as a little exception but the end result here is not pretty. So a few parts are getting consolidated in here. And the comment will go here to show why it's listed as a requirement.

> They conveniently don't use SCALE or BOOLEAN_ATTRIBUTES

That was supposed to mean "doesn't need formatting so the code is to display is easy". It's not really needed to mention that. Nevermind.